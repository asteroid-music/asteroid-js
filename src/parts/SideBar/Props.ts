//Imports from React module
import React from 'react';

/**
 * Interface for props for the <SideBar> React component.
 * Details for each individual property is given in comments below.
 */
interface SideBarProps {
    /** Name of the tab */
    tabName: string,
    /** List of tab names to generate buttons for */
    linkNames: string[],
    /** Whether to be open */
    open: boolean,
    /** Callback to call upon closing */
    closeCallback: ()=>void
}

export default SideBarProps
