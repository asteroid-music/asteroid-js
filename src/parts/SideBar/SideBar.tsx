//Import from external module 'react'
import React from 'react';

//Imports from external module '@material-ui/core'
import Drawer from '@material-ui/core/Drawer';
import Typography from '@material-ui/core/Typography';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';

//Imports from external module 'react-router'
import { Link } from 'react-router-dom';

//Imports from local folder
import SideBarProps from './Props';

/**
 * <SideBar> React component for navigating Asteroid.
 * Creates a bar at the side of the screen for routing links
 *
 * @param {`parts/SideBar/Props/SideBarProps`} props: the React props for the
 *      component
 */
function SideBar(props: SideBarProps) {
    return (
        <Drawer
            open={props.open}
            onClose={()=>{props.closeCallback();}}
        >
            <Typography variant="h5" align="center">{props.tabName}</Typography>
            <List style={{minWidth:"20vw"}}>
                {props.linkNames.map((subTabName: string) => {
                    return <ListItem
                        to={"/"+props.tabName+"/"+subTabName}
                        key={subTabName}
                        component={Link}
                        button
                        onClick={()=>{props.closeCallback();}}
                    >
                        <ListItemText primary={subTabName}/>
                    </ListItem>
                })}
            </List>
        </Drawer>
    );
}

export default SideBar;
