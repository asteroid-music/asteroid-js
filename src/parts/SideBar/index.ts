//Main component
export { default } from './SideBar';

//Props interface
export { default as SideBarProps } from './Props'

