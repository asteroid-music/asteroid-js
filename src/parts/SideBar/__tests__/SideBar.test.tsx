import React from 'react';
import renderer from 'react-test-renderer';
import SideBar from 'parts/SideBar';
import { MemoryRouter } from 'react-router-dom';

describe('<SideBar />',()=>{
    //Basic rendering with minimal props
    describe('basic rendering',()=>{
        it("renders correctly",()=>{
            expect(renderer
                .create(
                <MemoryRouter initialEntries={["/"]}>
                    <SideBar
                        tabName="openTab"
                        open
                        linkNames={["link1","link2"]}
                        closeCallback={()=>{}}
                    />
                </MemoryRouter>
                )
                .toJSON()
            ).toMatchSnapshot();
        });
    });
});