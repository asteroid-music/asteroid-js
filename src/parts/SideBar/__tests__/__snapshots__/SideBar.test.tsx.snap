// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SideBar /> basic rendering renders correctly 1`] = `
<Material UI Drawer
  onClose={[Function]}
  open={true}
>
  <Material UI Typography
    align="center"
    variant="h5"
  >
    openTab
  </Material UI Typography>
  <Material UI List
    style={
      Object {
        "minWidth": "20vw",
      }
    }
  >
    <Material UI ListItem
      button={true}
      component={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "displayName": "Link",
          "propTypes": Object {
            "innerRef": [Function],
            "onClick": [Function],
            "replace": [Function],
            "target": [Function],
            "to": [Function],
          },
          "render": [Function],
        }
      }
      onClick={[Function]}
      to="/openTab/link1"
    >
      <Material UI ListItemText
        primary="link1"
      />
    </Material UI ListItem>
    <Material UI ListItem
      button={true}
      component={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "displayName": "Link",
          "propTypes": Object {
            "innerRef": [Function],
            "onClick": [Function],
            "replace": [Function],
            "target": [Function],
            "to": [Function],
          },
          "render": [Function],
        }
      }
      onClick={[Function]}
      to="/openTab/link2"
    >
      <Material UI ListItemText
        primary="link2"
      />
    </Material UI ListItem>
  </Material UI List>
</Material UI Drawer>
`;
